llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -106,11 +106,35 @@ bool PostPipeliner::canAccept(MachineBasicBlock &LoopBlock) {
   return true;
 }
 
+int PostPipeliner::getResMII(MachineBasicBlock &LoopBlock) {
+  // For each instruction, find the first cycle in which it fits and collect the
+  // maximum
+  std::vector<uint64_t> Scoreboard(NInstr, 0);
+  int MII = 1;
+  for (auto &MI : LoopBlock) {
+    auto *SlotInfo = TII->getSlotInfo(TII->getSlotKind(MI.getOpcode()));
+    SlotBits Slots = SlotInfo ? SlotInfo->getSlotSet() : 0;
+
+    int C = 0;
+    while (C < NInstr && (Scoreboard[C] & Slots)) {
+      C++;
+    }
+    if (C >= NInstr) {
+      MII = NInstr;
```

Comment:
When this happens, do we try to run the pipeliner anyway? Should we return `PostPipelinerMaxII` (provides share this option)?

Reply:
This computes ResMII, not how it it used. I don't want to tie too many strings to it.
As we compute it now, it will never exceed NInstr. but even if II == NInstr, the latency path through those NInstrs can allow a 4 stage pipeline to be useful. And if the DAG is a straight chain of dependences, pipelining can still be cheap.